src/scanners/base.js

```diff
@@ -18,11 +18,16 @@ export default class BaseScanner {
     return 'string';
   }
 
+  static get scannerName() {
+    throw new Error('scannerName is not implemented');
```

Comment:
This is modelled after how `_getContents` works, also since this is a static method I can't check this properly in the constructor since that'd already be too late.